Added database functionality and several components

bruceferguson3 opened the pull request:
- Added 2 components, 
1) Title, mini desc, price, review Avg, review counter
2) Quantity, quantity counter
- Added 10 items to database to test functionality.
- Product can be changed in server.js by changing the query to id = 1-10

bruceferguson3 committed:
File Structure made.

bruceferguson3 committed:
Title, small description, price, stars, review Avg, and review count all structured properly.

bruceferguson3 committed:
DateBase tested with 10 items. Information switches out perfect with each load of the page. Added a quantity bar with no functionality yet.

mfuechec commented on webpack.config.js:
```diff
@@ -0,0 +1,19 @@
+        rules : [
+            {
+                test : /\.js?/,
+                include : SRC_DIR,
```
exclude node_modules

mfuechec replied:
Exclude node_modules

mfuechec commented on client/App.js:
```diff
@@ -1,32 +1,61 @@
 import React from 'react'
+import axios from 'axios'
 import TopPackage from "./Components/TopPackage.jsx";
+import MiddlePackage from "./Components/MiddlePackage.jsx";
```
More descriptive naming, maybe?

mfuechec commented on client/App.js:
```diff
@@ -1,32 +1,61 @@
+                                     productColorOpt: productInfo.data[0].color})
+            })
+            .catch(() => {
+                console.log('NO PRODUCT INFO')
```
Might want to pass in the error and log it. Could be other problems besides 'no info found'

mfuechec commented on client/Components/MiddlePackage.jsx:
```diff
@@ -0,0 +1,21 @@
+    return (
+
+        <div className='middlePackage'>
+            <span className='quantityTitle'>Quantity</span>
```
Css naming convention would be quantity-title. Check it out, gurl:
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

mfuechec commented on server/db.js:
```diff
@@ -0,0 +1,33 @@
+const connection = mysql.createConnection({
+    host: 'localhost',
+    user: 'root',
+    password: 'password',
```
You can put your password in a separate file that is gitignore-d, then reference it here.

mfuechec reviewed (changes requested):
I think you might want to hide your password before it gets pulled, but that's up to you.

bruceferguson3 committed:
Add 10 more products to database.
Add 'onSale' and 'new' column to database.
Add conditional rendering to my components.

bruceferguson3 committed:
Add styling to the 'new' and 'onSale Price' components

bruceferguson3 committed:
Add password to git ignore

bruceferguson3 committed:
Database has 30 products.
Add extra columns such as dealLen and regPrice for sales with 'ikeaFamilySale'
Add 3 new options such as mattress, legs, and slatted bed base

bruceferguson3 committed:
Fix former pull request issues

bruceferguson3 committed:
Made 3rd component
Fix several issues with database
Add css to improve the look and positioning of elements
All database product columns are set and ready to be filled with the rest of the product information

bruceferguson3 committed:
Add new color option and button to open modal options

bruceferguson3 committed:
Add modal
Add 20 new products to database

bruceferguson3 committed:
Database is FINALLY completed.
